Plugins/NativeEngine/Source/NativeEngine.h

```diff
@@ -243,7 +244,48 @@ namespace Babylon
 
         std::vector<Napi::FunctionReference> m_requestAnimationFrameCallbacks{};
 
+        template<typename T>
+        class SingleFrameValue
+        {
+        public:
+            SingleFrameValue(GraphicsImpl& impl, arcana::cancellation_source& cancellation, T defaultValue)
+                : m_impl{impl}
+                , m_cancellationSource{cancellation}
+                , m_defaultValue{defaultValue}
+                , m_value{defaultValue}
+                , m_isResetScheduled{false}
+            {
+            }
+            
+            T Get(bgfx::Encoder&) const
+            {
+                return m_value;
+            }
+
+            void Set(bgfx::Encoder&, bool value)
+            {
+                m_value = value;
+                if (!m_isResetScheduled)
+                {
+                    arcana::make_task(m_impl.AfterRenderScheduler(), m_cancellationSource, [this]() {
+                        m_value = m_defaultValue;
+                        m_isResetScheduled = false;
+                    });
+                    m_isResetScheduled = true;
+                }
+            }
+
+        private:
+            GraphicsImpl& m_impl;
+            arcana::cancellation_source& m_cancellationSource;
+            const T m_defaultValue{};
+            T m_value{};
+            bool m_isResetScheduled{};
+        };
+
         const VertexArray* m_boundVertexArray{};
+        FrameBuffer m_defaultFrameBuffer;
         FrameBuffer* m_boundFrameBuffer{};
+        SingleFrameValue<bool> m_boundFrameBufferNeedsRebinding;
```

Comment:
While this is the only one at the moment, frame-temporal values have occurred several times; with a slightly different implementation, the concept extends to things like ViewIds, whether or not certain operations have been scheduled, etc. I'm anticipating this being used in more places. For clarity, though, I've moved it to its own file and added comments in response to these comments.

Reply:
Regarding `m_boundFrameBufferNeedsRebinding`, this happens when the frame changes and Babylon attempts to "leave" FrameBuffer bound across frames. Because ViewIds are now assigned on binding, "leaving" the FrameBuffer bound across a frame boundary is no longer supported, so it must be unbound and rebound under these circumstances to get a new ViewId. This is likely to evolve as our understanding of the relationship between the FrameBuffers and the Context develops.